code/game/objects/items/grenades/chem_grenade.dm

```diff
@@ -113,11 +113,11 @@
 			stage_change(GRENADE_READY)
 			to_chat(user, span_notice("You lock the [initial(name)] assembly."))
 			tool.play_tool_sound(src, 25)
-		if(landminemode)
-			landminemode.timing = FALSE
-			if(!landminemode.secured)
-				landminemode.toggle_secure()
-			landminemode.toggle_scan(FALSE)
+			if(landminemode)
```

Comment:
wouldn't it be better to move this past the else? We (I think) want to be able to toggle timing and secures without locking the grenade

Reply:
i might be misunderstanding something, but i'm pretty sure there's no way to toggle timings without locking an unlocked grenade, because both use screwdriver_act right now

Reply:
Excuse me uh.
As things currently stand, you can change landmine stuff without the grenade needing to have beakers inside it. This changes that.
Rather then nesting this in a level, it seems like it would make more sense for it to be above the if(beakers.len) check, i'd think.

Reply:
sorry for taking so long; i've found a solution which looks pretty good to me and seems like what the original code author meant to do. i'd appreciate it if you could take a look

Reply:
(also it lets players change the landmine stuff without needing beakers, like you said)

Reply:
to be just brutally honest with you I have no idea what the first guy was trying to do. This seems more functional to me though